### ISSUE TITLE
networkx 1.11: TypeError: 'method' object is not subscriptable

### ISSUE THREAD

**Issue by hroncok** [mikedh/trimesh#213]
I'm getting this failure with networkx 1.11 (available in Fedora):
```
    def to_edgelist(self):
        """
            Export the current transforms as a list of edge tuples, with
            each tuple having the format:
            (node_a, node_b, {metadata})
    
            Returns
            -------
            edgelist: (n,) list of tuples
            """
        # save cleaned edges
        export = []
        # loop through (node, node, edge attributes)
        for edge in nx.to_edgelist(self.transforms):
            a, b, c = edge
            # geometry is a node property but save it to the
            # edge so we don't need two dictionaries
>           if 'geometry' in self.transforms.nodes[b]:
E           TypeError: 'method' object is not subscriptable
```

I'd say `self.transforms.nodes` is `networkx.DiGraph.nodes` and that is a method in 1.11 and property in 2?

See also https://github.com/gboeing/osmnx/issues/91

I'll try to update networkx to 2, but if indeed 2 is needed, please indicate so in https://github.com/mikedh/trimesh/blob/b9b139e89bbd9cdc86c837cb84ce261bf4a77b9f/setup.py#L49

**Comment by hroncok** [comment]
I was able to workaround this by:

```diff
diff --git a/trimesh/scene/transforms.py b/trimesh/scene/transforms.py
--- a/trimesh/scene/transforms.py
+++ b/trimesh/scene/transforms.py
@@ -145,8 +145,13 @@ class TransformForest:
             a, b, c = edge
             # geometry is a node property but save it to the
             # edge so we don't need two dictionaries
-            if 'geometry' in self.transforms.nodes[b]:
-                c['geometry'] = self.transforms.nodes[b]['geometry']
+            try:
+                nodes_b = self.transforms.nodes[b]
+            except TypeError:
+                # networkx 1.11
+                nodes_b = self.transforms.node[b]
+            if 'geometry' in nodes_b:
+                c['geometry'] = nodes_b['geometry']
             # save the matrix as a float list
             c['matrix'] = np.asanyarray(c['matrix'],
                                         dtype=np.float64).tolist()
```

Would this be acceptable as a PR?

**Comment by mikedh** [comment]
Hey, thanks for the report! I intended it to work with both 1.x/2.x versions of networkx, but it looks like I haven't tested on 1.x in some time. I'll add it to Travis.

Looking at the [migration guide](https://networkx.github.io/documentation/stable/release/migration_guide_from_1.x_to_2.0.html) it seems like for now replacing `.nodes` with `.node` is the suggested solution (at least until they remove it in 3.x 🙄) It works for me on both 1.11 and 2.x. I pushed that change as well as adding 1.11 to Travis in https://github.com/mikedh/trimesh/commit/89fb72fc40373ecc77042381f087a24083b8c045
